Source/JavaScriptCore/runtime/JSGlobalObject.h

```diff
@@ -623,6 +624,8 @@ class JSGlobalObject : public JSSegmentedVariableObject {
 
     const Ref<ImportMap> m_importMap;
 
+    const Ref<SpeculationRules> m_speculationRules;
```

Comment:
I would be surprised if JavaScriptCore needs to know anything about SpeculationRules.  Can this be moved to LocalDOMWindow or JSDOMGlobalObject?

Reply:
Moved the entire class to WebCore